### PR TITLE
Fix #4637: Support JWT Token generation for bot accounts

### LINKED ISSUE #4637: Support JWT Token generation for bot accounts
**Is your feature request related to a problem? Please describe.**

In OpenMetadata all APIs are secured and we need a authenticated JWT token to access APIs. 
There are two type of user entities one represents a user who can visit the UI and click on SSO button to log themselves in and others are bots which represents automated workflows one such example of bot is our ingestion-bot. This bot requires a service account to authenticate and access the APIs.
However, not all SSO's support service accounts and some gives a site-wide access which can be security risk as the service account is only needed to access OpenMetadata APIs. Moreover provisioning a service account is cumbersome requires lot of steps to get it right.

To improve this use case we are going to create JWT token to authenticate a bot.

1. OpenMetadata server should be configured with RSA public/private key2. 
2. Administrator can create a bot account and will have option of choosing a SSO service account or JWT token as authenticationTokenMechanism   
4. Administrator can go to Bot management page and generate a JWT token
5. Administrator will have an option to set expiry limit from a day to unlimited 
7.  Administrator can use this JWT token to configure ingestion pipeline authentication mechanism and pass the JWT token 
9. It is up to the users to follow the best practices in keeping the JWT token secure and not leaked 
11. Administrators will have an option to revoke or disable the JWT token and generate a new token at any given point. 
13.  Currently, bots are at high privilege i.e they can create/update/read/delete operations on any entities. In future releases we can provide scopes on what a bot can access

### PR DESCRIPTION
**Pull request by harshach**
### Describe your changes :
see #4637 
#
### Type of change :

- [x] New feature

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: ingestion/src/metadata/ingestion/ometa/provider_registry.py [hunk 1]
```diff
@@ -65,3 +66,8 @@ def azure_auth_init(config: OpenMetadataConnection) -> AuthenticationProvider:
 @auth_provider_registry.add(AuthProvider.custom_oidc.value)
 def custom_oidc_auth_init(config: OpenMetadataConnection) -> AuthenticationProvider:
     return CustomOIDCAuthenticationProvider.create(config)
+
+
+@auth_provider_registry.add(AuthProvider.openmetadata.value)
+def om_auth_init(config: OpenMetadataConnection) -> AuthenticationProvider:
+    return OpenMetadataAuthenticationProvider.create(config)
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/ometa/provider_registry.py:73]: Hi @harshach , this LGTM. We'd just be missing the entry for the lineage backend here https://github.com/open-metadata/OpenMetadata/blob/main/ingestion/src/airflow_provider_openmetadata/lineage/config/providers.py

File: catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/UserRepository.java [hunk 2]
```diff
@@ -399,5 +404,28 @@ private void updateTeams(User origUser, User updatedUser) throws IOException {
...
+      AuthenticationMechanism origAuthMechanism = origUser.getAuthenticationMechanism();
+      AuthenticationMechanism updatedAuthMechanism = updatedUser.getAuthenticationMechanism();
+      if (origAuthMechanism == null && updatedAuthMechanism != null) {
+        recordChange(
+            "authenticationMechanism", origUser.getAuthenticationMechanism(), updatedUser.getAuthenticationMechanism());
+      } else if (origAuthMechanism != null
+          && updatedAuthMechanism != null
+          && origAuthMechanism.getConfig() != null
+          && updatedAuthMechanism.getConfig() != null) {
+        JWTAuthMechanism origJwtAuthMechanism =
```
Comment (sureshms) [inline catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/UserRepository.java:418]: We have && in the previous line no :)

File: ingestion/src/metadata/ingestion/ometa/provider_registry.py (hunk 1 above)
Comment (harshach) [reply ingestion/src/metadata/ingestion/ometa/provider_registry.py:73]: @pmbrull thanks for the catch. Fixed it.

File: catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/UserRepository.java (hunk 2 above)
Comment (harshach) [reply catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/UserRepository.java:418]: the first check is to set when the existing authMech is empty and the second check is to check if there is any change between the existing authMech and the new authMech. 
I'll re-read the code again tomorrow, might be missing something.

File: conf/openmetadata.yaml [hunk 3]
```diff
@@ -142,6 +143,12 @@ authenticationConfiguration:
   clientId: ${AUTHENTICATION_CLIENT_ID:-""}
   callbackUrl: ${AUTHENTICATION_CALLBACK_URL:-""}
 
+jwtTokenConfiguration:
```
Comment (akash-jain-10) [inline conf/openmetadata.yaml:146]: @harshach - Is there a configuration missing to provide location (path) for the below ?

Comment (harshach) [reply conf/openmetadata.yaml:146]: users are expected to configure absolute paths and this is optional configuration.

**Review by akash-jain-10 — APPROVED** [review]

**Pull request merged**